### PR TITLE
merge main after 1.58

### PR DESCRIPTION
**Pull request by lennyburdette**
---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-federation/src/supergraph/mod.rs [hunk 1]
```diff
@@ -1576,6 +1691,103 @@ lazy_static! {
     };
 }
 
+fn insert_directive(
+    schema: &mut FederationSchema,
+    pos: &CompositeTypeDefinitionPosition,
+    directive: Component<Directive>,
```
Comment (pubmodmatt) [inline apollo-federation/src/supergraph/mod.rs:1697]: Looks like a merge error here - the build is failing due to a missing `use` for `Directive`

Comment (lennyburdette) [reply apollo-federation/src/supergraph/mod.rs:1698]: yep, thanks. i had fixed it just forgot to commit it!

**Review by pubmodmatt — APPROVED** [review]

**Review by garypen — CHANGES_REQUESTED** [review]
Please don't merge this PR. I'm about to merge `dev` into `next`. See: https://github.com/apollographql/router/pull/6382

I need to understand why we should merge `main` to `next` rather than `dev` as we have been so far.

**Comment by abernix** [comment]
This is my fault.  I suggested to @lennyburdette  that `main` might be mildly less volatile than `dev` to merge in prior to a release on the `next` branch (a 2.0.0-preview).   I didn't think it would cause more problems since they all share a branch strategy, but if we are regularly merging `dev` into `next` already then using `main` literally doesn't make sense anyhow.

Maybe sort this out on the agenda of today's group huddle and I'll keep out of the way next time. 😄

**Pull request closed without merging**